### PR TITLE
Add API for `sf::WindowBase`

### PR DESCRIPTION
**Pull request by ChrisThrasher**
Depends on #206

### CODE REVIEW

**Comment by Marioalexsan** [comment]
The code is missing a binding for `createVulkanHandle`.

**Comment by ChrisThrasher** [comment]
> The code is missing a binding for `createVulkanHandle`.

Thanks, should be done now

File: include/SFML/Window/WindowBase.h [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+/// \param enabled    sfTrue to enable, sfFalse to disable
+///
+////////////////////////////////////////////////////////////
+CSFML_WINDOW_API void sfWindowBase_setKeyRepeatEnabled(sfWindowBase* windowBase, sfBool enabled);
+
+////////////////////////////////////////////////////////////
+/// \brief Change the joystick threshold
+///
+/// The joystick threshold is the value below which
+/// no JoyMoved event will be generated.
```
Comment (eXpl0it3r) [inline include/SFML/Window/WindowBase.h:322]:
```suggestion
/// no JoystickMoved event will be generated.
```

It's also wrong in the Window class

File: include/SFML/Window/WindowBase.h [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+////////////////////////////////////////////////////////////
+/// \brief Request the current window to be made the active
+/// foreground window
+///
+/// At any given time, only one window may have the input focus
+/// to receive input events such as keystrokes or mouse events.
+/// If a window requests focus, it only hints to the operating
+/// system, that it would like to be focused. The operating system
+/// is free to deny the request.
+/// This is not to be confused with sfWindowBase_setActive().
```
Comment (eXpl0it3r) [inline include/SFML/Window/WindowBase.h:339]: WindowBase has no `setActive`, it's also a copy paste error in SFML itself

File: src/SFML/Window/WindowBase.cpp [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+
+////////////////////////////////////////////////////////////
+sfBool sfWindowBase_pollEvent(sfWindowBase* windowBase, sfEvent* event)
+{
+    CSFML_CHECK_RETURN(windowBase, sfFalse);
+    CSFML_CHECK_RETURN(event, sfFalse);
+
+    // Get the event
+    sf::Event SFMLEvent;
```
Comment (eXpl0it3r) [inline src/SFML/Window/WindowBase.cpp:102]:
```suggestion
    sf::Event sfmlEvent;
```

I know this is just copy-pasted

File: src/SFML/Window/WindowBase.cpp [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+
+////////////////////////////////////////////////////////////
+sfBool sfWindowBase_waitEvent(sfWindowBase* windowBase, sfEvent* event)
+{
+    CSFML_CHECK_RETURN(windowBase, sfFalse);
+    CSFML_CHECK_RETURN(event, sfFalse);
+
+    // Get the event
+    sf::Event SFMLEvent;
```
Comment (eXpl0it3r) [inline src/SFML/Window/WindowBase.cpp:123]:
```suggestion
    sf::Event sfmlEvent;
```

File: include/SFML/Window/Vulkan.h [hunk 5]
```diff
@@ -28,6 +28,21 @@
 #include <SFML/Window/Export.h>
 
 
+typedef struct VkInstance_T* VkInstance;
```
Comment (eXpl0it3r) [inline include/SFML/Window/Vulkan.h:31]: Where is `VkInstance_T` defined?

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Some small changes, then this should be ready to go

File: include/SFML/Window/Vulkan.h (hunk 5 above)
Comment (ChrisThrasher) [reply include/SFML/Window/Vulkan.h:31]: This is the line where's defined. I copied this out of SFML itself. I think it's defining a struct named `VkInstance_T` then immediately defining a pointer to that struct named `VkInstance`.

Comment (ChrisThrasher) [reply include/SFML/Window/Vulkan.h:31]: Wait sorry I see what you mean now. This is just a forward declaration. Good question. I'm not sure where it's defined as a matter of fact.

Comment (ChrisThrasher) [reply include/SFML/Window/Vulkan.h:31]: https://android.googlesource.com/platform/frameworks/native/+/b1352bc/vulkan/libvulkan/loader.cpp#35

I found it defined inside some Vulkan stuff. Not sure if this is an authoritative source but it seems to imply that this structure is something defined by Vulkan.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**